base/bucket_n1ql.go

```diff
@@ -124,6 +124,19 @@ func (bucket *CouchbaseBucketGoCB) BuildDeferredIndexes(indexSet []string) error
 	return BuildDeferredIndexes(bucket, indexSet)
 }
 
+func (bucket *CouchbaseBucketGoCB) waitUntilQueryServiceReady(_ time.Duration) error {
```

Comment:
Is my understanding that only under `gocb/v1` that we use this code?

Reply:
Yep

Reply:
I'm about to pull out the https://cs.github.com/couchbase/sync_gateway/blob/e94af5ecfb6f85e9e0b4b5c91a7675560a343f99/base/util_testing.go#L663 because it won't work with collections - and this code then won't need to be modified. I tried to pull the whole file (bucket_n1ql.go) but there are some constants that are used by new files and it was a mess that I will defer.

You could remove that right now and then we will leave base/bucket_n1ql.go / base/bucket.go mostly unused, or we can preserve this for future.

Reply:
Is there a ticket to track removal of gocbv1 code? If not, filing that and getting the information in there would be good.

Reply:
https://issues.couchbase.com/browse/CBG-2339